ComfortZoneApp/app/src/main/java/com/example/comfortzone/profile/ui/ProfileFragment.java

```diff
@@ -82,8 +87,27 @@ private void setUpRecyclerView() {
     }
 
     private void getSavedCities() {
-        ArrayList<Integer> savedCityIds = (ArrayList<Integer>) currentUser.get(KEY_SAVED_CITIES);
-        AllWeathersDatabase db = AllWeathersDatabase.getDbInstance(getContext());
-        adapter.addAll(savedCityIds.stream().map(cityId -> db.weatherDao().getWeatherById(cityId)).collect(Collectors.toList()));
+        Observable<Object> dataSetupObservable = WeatherDbUtil.maybeUpdateCitiesList(getActivity());
```

Comment:
I see..this is what you were saying, hopefully setting these up in HostingActivity wouldn't get too complicated